### PR TITLE
[Fixes #59] Centralizing error handler structure

### LINKED ISSUE #59: Centralize error handling
As mentioned in #57, our error handling is a weak point.

Here we propose the creation of a centralized error handling for the application, that will be the default error handling of every route. Anywhere inside the routes, we can throw a custom error, in a standard format defined by us, and the error handler will be in charge of returning a response to the user.

For errors that do not imply in a bad response to the user, the try-catch can be used in inside the route.

### PR DESCRIPTION
**Pull request by tloriato**
- Created a file for our custom errors, that extend a BaseError class.
- Created a module to register the error handlers.
- Updated invoice endpoints so that they are compatible with new architecture.
- Updated blocks endpoints so that they are compatible with new architecture.

### CODE REVIEW

File: api/src/services/users/index.js [hunk 1]
```diff
@@ -1,64 +1,75 @@
...
+    })
+    .catch((err) => {
+      throw err;
     });
-    resolve({ code: 500, data: { errors } });
-  }
-});
+};
+
+//   } catch (err) {
```
Comment (tloriato) [inline api/src/services/users/index.js]: Podia não ter os comments?

File: api/src/services/invoices/index.js [hunk 2]
```diff
@@ -53,104 +54,94 @@ const listInvoices = async req => new Promise(async (resolve) => {
...
-      throw new Error('Esse usuário não pertence à uma empresa, logo não pode emitir notas fiscais');
-    }
+  const invoiceInfo = serializers.invoice.deserialize(req.body);
+
+  // setting enderecoEmissor
+  const empresa = await models.empresa.findByPk(user.empresaCnpj, { raw: true });
+  invoiceInfo.enderecoEmissor = empresa.enderecoBlockchain;
+  // setting blocoConfirmacaoId
+  const lastBlock = await models.block.findOne({ raw: true });
+  invoiceInfo.blocoConfirmacaoId = lastBlock.block_id;
```
Comment (tloriato) [inline api/src/services/invoices/index.js:91]: Nem esses?

File: api/src/services/users/index.js (hunk 1 above)
Comment (dbeyda) [reply api/src/services/users/index.js]: Opa, tirei!

File: api/src/services/invoices/index.js (hunk 2 above)
Comment (dbeyda) [reply api/src/services/invoices/index.js:91]: tirado

**Pull request merged**